### PR TITLE
Add the changes in the changelog for the project make

### PR DESCRIPTION
**Pull request by sbrunner**
Update some documentation

### CODE REVIEW

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 1]
```diff
@@ -4,11 +4,21 @@ This file includes migration steps for each release of c2cgeoportal.
 Version 1.6
 ===========
 
-1. Get the new files:
+1. Get the new and heavy modified files:
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: heavy => heavily

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 2]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
-
-3. In the {{package}}/__init__.py file do the following changes:
+    git rm -r {{package}}/static/build
+    git rm config.yaml
+    git rm bootstrep.py
+    git rm -r *buildout*
+    git rm development.ini.in production.ini.in
+    git rm versions.cfg
+
+3. In this version we use a new build system based on make that will change
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: make => GNU Make

Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: "In this version" => Starting from version 1.6

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 3]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+   many of your habits, than read, than read:
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: "than read, than read" => Information are given at:

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 4]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+   http://docs.camptocamp.net/c2cgeoportal/1.6/integrator/make.html
+
+   Main things to do:
+   - Move your old `config.yaml` and your `vars` section of the old
+     `buildout.cfg` file to the new `{{package}}.yaml` file.
+   - Move your special task from your old `buildout.cfg` file to your new
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: What do you call "task"?

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 5]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+     `{{package}}.mk` file
+   - The `README.rst`, `.gitignore` files will be overwritten have a look if
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: The `README.rst`, `.gitignore` files will be overwritten. Please make sure to back your changes up before proceeding.

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 6]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+     you should get some things back
+   - The `development.ini.in`, `production.ini.in` files will be replaced by
+     the `development.ini.mako`, `production.ini.mako` files, If you lost some
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Replace "," by "." (period) at the end of "`production.ini.mako` files,"

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 7]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+     thing it should probably be in the `{{package}}.yaml` file.
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: some thing => something
it should probably => it is probably (the sentence is not very clear though)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 8]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+   - The `print/template'print.mako` file should be moved to
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: print/template'print.mako => print/templates/print.mako
(are talking about "print.mako.in" and "print.yaml.mako.in"?)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 9]
```diff
@@ -16,37 +26,78 @@ Version 1.6
...
+     `print.yaml.mako`.
+   - All the build environement yhould be created again, for multi project
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: The build environment must be recreated [which means??]. For the multi-project mode...

File: doc/integrator/make.rst [hunk 10]
```diff
@@ -0,0 +1,83 @@
+.. _integrator_make:
+
+Build the project with make
```
Comment (asaunier) [inline doc/integrator/make.rst]: make => GNU Make (or at leat Make with the capital M)

File: doc/integrator/make.rst [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+===========================
+
+Makefiles
+---------
+
+Usually we have the following makefiles includes:
+``<user>.mk`` -> ``<package>.mk`` -> ``CONST_Makefile.mk``.
+
+The ``CONST_Makefile.mk`` is a huge makefile that maintain by the
```
Comment (asaunier) [inline doc/integrator/make.rst]: that maintain => that is maintained

File: doc/integrator/make.rst [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+c2cgeoportal developers.
```
Comment (asaunier) [inline doc/integrator/make.rst]: c2cgeoportal => GeoMapFish (?)

File: doc/integrator/make.rst [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+The ``<package>.mk`` contains the project specific things,
```
Comment (asaunier) [inline doc/integrator/make.rst]: ``<package>.mk`` contains the project-specific config. Default is:

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 4 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Should parts for buildout and rule for make: https://www.gnu.org/software/make/manual/make.html#Rule-Example
=> ?
Move your special buildout parts from your old `buildout.cfg` file as a make rule in your new `{{package}}.yaml` file.

File: doc/integrator/make.rst [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    VARS_FILES += ${VARS_FILE} vars_<package>.yaml
+    else
+    VARS_FILE = vars_<package>.yaml
+    VARS_FILES += ${VARS_FILE} 
+    endif
+
+    include CONST_Makefile
+
+
+And the ``<user>.mk`` contains the user specific things, mainly the
```
Comment (asaunier) [inline doc/integrator/make.rst]: ``<user>.mk`` contains the user-specific config (mainly the ``INSTANCE_ID`` config) and should look like:

File: doc/integrator/make.rst [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    DEVELOPMENT = TRUE
+
+    include <package>.mk
+
+
+Vars files
+----------
+
+We also have some files that describe the variables of the project:
+``vars_<package>.yaml`` that extends the ``CONST_vars.yaml``.
```
Comment (asaunier) [inline doc/integrator/make.rst]: The project variables are set in the ``vars_<package>.yaml`` file, which extends the default ``CONST_vars.yaml``.

File: doc/integrator/make.rst [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+For more informations see the
```
Comment (asaunier) [inline doc/integrator/make.rst]: informations => information ("information" is generally uncountable)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 7 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: missing word 'moved' => ?
... something it should probably be moved in the `{{package}}.yaml` file.

File: doc/integrator/make.rst [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+`c2c.template <https://github.com/sbrunner/c2c.template>`_ documentation.
+
+
+Makefile config valiables
```
Comment (asaunier) [inline doc/integrator/make.rst]: valiables => variables

File: doc/integrator/make.rst [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+-------------------------
+
+In the makefiles won can set the following variables:
```
Comment (asaunier) [inline doc/integrator/make.rst]: The following variables may be set in the makefiles:

File: doc/integrator/make.rst [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+* ``DEVELOPMENT``: ``TRUE`` to don't minify the css, js, defaults to ``FALSE``.
```
Comment (asaunier) [inline doc/integrator/make.rst]: if ``TRUE`` CSS and JS files are not minified. Defaults to ``FALSE``.

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 8 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: no, print.mako.in and  print.yaml.mako (I miss one .in :-) )

File: doc/integrator/make.rst [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+* ``LANGUAGES``: the list of available languages.
+* ``CGXP``: ``TRUE`` to build the CGXP interface, defaults to ``TRUE``.
+* ``MOBILE``: ``TRUE`` to build the Sencha touch interface, defaults to ``TRUE``.
```
Comment (asaunier) [inline doc/integrator/make.rst]: touch => Touch

File: doc/integrator/make.rst [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+* ``NGEO``: ``TRUE`` to build the ngeo interface, defaults to ``FALSE``.
+* ``TILECLOUD_CHAIN``: ``TRUE`` to indicate that we use TileCloud-chain, defaults to ``TRUE``.
+* ``PRINT2``: , ``TRUE`` to build the print 2 server, defaults to ``FALSE``.
+* ``PRINT3``: , ``TRUE`` to build the print 2 server, defaults to ``TRUE``.
```
Comment (asaunier) [inline doc/integrator/make.rst]: print 2 => print 3
(what about using the full "MapFish Print 3" name?)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 9 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: => ?
The build environments (`<user>.mk` files) must be recreated again.

File: doc/integrator/make.rst (hunk 21 above)
Comment (asaunier) [inline doc/integrator/make.rst]: It is relevent to have both config ``PRINT2`` and ``PRINT3``? What happens if you set both to TRUE?

File: doc/integrator/make.rst [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+* ``PRE_RULES``: predefine some build rules, default is empty.
+* ``POST_RULES``: postdefine some build rules, default is empty.
+* ``DISABLE_BUILD_RULES``: List of rules we want to disable, default is empty.
+
+
+Custom rule
+-----------
+
+In the ``<package>.mk`` file we can create some other rules, simple example
```
Comment (asaunier) [inline doc/integrator/make.rst]: some other rules, simple example => some other rules. Here is a simple example that copies a file:

File: doc/integrator/print.rst [hunk 23]
```diff
@@ -105,16 +92,19 @@ And in ``print/templates/A3_landscape_inherit.mako`` and
 ``print/templates/A4_portrait_inherit.mako`` thoses block will
 be redefined.
 
-The ``print.mako`` has the "header" part and includes the wanted templates.
+The ``print.yaml.mako`` has the "header" part and includes the wanted templates.
```
Comment (asaunier) [inline doc/integrator/print.rst]: ``print.yaml.mako.in``? (are there ".in" files with the new Make system?)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 4 above)
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: OK

File: doc/integrator/make.rst (hunk 21 above)
Comment (sbrunner) [inline doc/integrator/make.rst]: probably not but booth to FALSE yes :-)

It's true that can be replaces by a PRINT_VERSION with the value not in [2, 3] => desable print

File: doc/integrator/print.rst (hunk 23 above)
Comment (sbrunner) [inline doc/integrator/print.rst]: Yes but mako do the same thaw now we interpret it only one time :-)

**Pull request merged**